cmd/auth/auth.go

```diff
@@ -31,13 +31,21 @@ var (
 	password string
 	baseURL  string
 	model    string
+	engine   string
 )
 
 // authCmd represents the auth command
 var AuthCmd = &cobra.Command{
 	Use:   "auth",
 	Short: "Authenticate with your chosen backend",
 	Long:  `Provide the necessary credentials to authenticate with your chosen backend.`,
+	PreRun: func(cmd *cobra.Command, args []string) {
+		backend, _ := cmd.Flags().GetString("backend")
+		if backend == "azureopenai" {
```

Comment:
This might sound pedantic, but I would advise `strings.ToLower(backend)` on that evaluation